File: Remi_Hasan/src/main/java/qls/model/Page.java
package qls.model;

import qls.model.statement.Statement;
import qls.visitor.IQLSVisitor;

import java.util.List;

public class Page extends QLSNode {

    private final String identifier;
    private final List<Statement> statements;

    public Page(String identifier, List<Statement> statements) {
        this.identifier = identifier;
        this.statements = statements;
    }

    public String getIdentifier() {
        return identifier;
    }

    public List<Statement> getStatements() {
        return statements;
    }

    @Override
    public <T> T accept(IQLSVisitor<T> visitor) {
        return visitor.visit(this);
    }
}

File: Erik_Vincent/Assignment1/Assignment1/Model/QLS/AST/Page.cs
using Assignment1.Model.QL.AST;
using System.Collections.Generic;
using System.Linq;

namespace Assignment1.Model.QLS.AST
{
    public class Page : ASTNode, IQLSASTNode
    {
        public string Id { get; }
        public IReadOnlyCollection<Statement> Statements => _statements.AsReadOnly();

        private readonly List<Statement> _statements;

        public Page(int lineNumber, string id, IEnumerable<Statement> statements)
        {
            _lineNumber = lineNumber;
            Id = id;
            _statements = statements.ToList();
        }

        public void Accept(IQLSASTVisitor visitor) => visitor.Visit(this);
    }
}
